### PR TITLE
Added user sessions

### PR DESCRIPTION
**Pull request by tivvit**

### CODE REVIEW

**Comment by tivvit** [comment]
After #24

File: server/database/users.go [hunk 1]
```diff
@@ -3,21 +3,23 @@ package database
...
 	_, err := db.Exec(`insert into d_user (name, password) values (?, ?)`, username, password)
 	return err
 }
 
-func GetUserPassword(db *sql.DB, username string) (string, error) {
+func GetUserPassword(db *sql.DB, username string) (stycoon.PlayerId, string, error) {
 	var password sql.NullString
-	err := db.QueryRow("select password from d_user where name = ? limit 1", username).Scan(&password)
+	var player stycoon.PlayerId
+	err := db.QueryRow("select id, password from d_user where name = ? limit 1", username).Scan(&player.Id, &password)
```
Comment (evamayerova) [inline server/database/users.go:17]: This returns user id, not player id. You probably want to create a separate method to query player id from t_player table based on the player name from the request.

File: server/handlers/handlers.go [hunk 2]
```diff
@@ -3,11 +3,12 @@ package handlers
 import (
 	"database/sql"
 	"encoding/json"
-	"io/ioutil"
-	"net/http"
-
+	"fmt"
 	"github.com/gdg-garage/space-tycoon/server/stycoon"
+	"github.com/gorilla/sessions"
```
Comment (evamayerova) [inline server/handlers/handlers.go:8]: 🦍

File: server/stycoon/users.go [hunk 3]
```diff
@@ -12,3 +18,32 @@ func HashPassword(password string) (string, error) {
...
+	if username, ok := session.Values["username"]; ok {
+		if username, ok = username.(string); ok {
+			user.Name = username.(string)
+		} else {
+			return Credentials{}, errors.New("username is corrupted = user is not logged")
+		}
+	} else {
+		return Credentials{}, errors.New("username is missing = user is not logged")
+	}
+	if username, ok := session.Values["player"]; ok {
```
Comment (evamayerova) [inline server/stycoon/users.go:37]: nit: playername

File: server/handlers/handlers.go (hunk 2 above)
Comment (tivvit) [reply server/handlers/handlers.go:8]: :joy:

File: server/database/users.go (hunk 1 above)
Comment (tivvit) [reply server/database/users.go:17]: yeah this needs more work - we should finish it in this MR - I will prepare it tomorrow ;)

Comment (tivvit) [reply server/database/users.go]: done

**Pull request merged**